warehouse/integrations/redshift/redshift_test.go

```diff
@@ -801,584 +803,595 @@ func TestIntegration(t *testing.T) {
+
+			loadFiles := []whutils.LoadFile{{Location: uploadOutput.Location}}
+			mockUploader := newMockUploader(t, loadFiles, tableName, whutils.DiscardsSchema, whutils.DiscardsSchema, whutils.LoadFileTypeCsv)
+
+			rs := redshift.New(config.New(), logger.NOP, stats.NOP)
+			err := rs.Setup(ctx, warehouse, mockUploader)
+			require.NoError(t, err)
+
+			err = rs.CreateSchema(ctx)
+			require.NoError(t, err)
+
+			err = rs.CreateTable(ctx, tableName, whutils.DiscardsSchema)
+			require.NoError(t, err)
+
+			loadTableStat, err := rs.LoadTable(ctx, tableName)
+			require.NoError(t, err)
+			require.Equal(t, loadTableStat.RowsInserted, int64(6))
+			require.Equal(t, loadTableStat.RowsUpdated, int64(0))
+
+			records := whth.RetrieveRecordsFromWarehouse(
+				t,
+				rs.DB.DB,
+				fmt.Sprintf(`
+					SELECT
+					  column_name,
+					  column_value,
+				      reason,
+					  received_at,
+					  row_id,
+					  table_name,
+					  uuid_ts
+					FROM
+					  %q.%q
+					ORDER BY
+					  row_id ASC;
+				`,
+					warehouse.Namespace,
+					tableName,
+				),
+			)
+			require.Equal(t, records, whth.DiscardTestRecords())
+		})
+		t.Run("parquet", func(t *testing.T) {
+			ctx := context.Background()
+			tableName := "parquet_test_table"
+
+			uploadOutput := whth.UploadLoadFile(t, fm, "../testdata/load.parquet", tableName)
+
+			fileStat, err := os.Stat("../testdata/load.parquet")
+			require.NoError(t, err)
+
+			loadFiles := []whutils.LoadFile{{
+				Location: uploadOutput.Location,
+				Metadata: json.RawMessage(fmt.Sprintf(`{"content_length": %d}`, fileStat.Size())),
+			}}
+			mockUploader := newMockUploader(t, loadFiles, tableName, schemaInUpload, schemaInUpload, whutils.LoadFileTypeParquet)
+
+			rs := redshift.New(config.New(), logger.NOP, stats.NOP)
+			err = rs.Setup(ctx, warehouse, mockUploader)
+			require.NoError(t, err)
+
+			err = rs.CreateSchema(ctx)
+			require.NoError(t, err)
+
+			err = rs.CreateTable(ctx, tableName, schemaInUpload)
+			require.NoError(t, err)
+
+			loadTableStat, err := rs.LoadTable(ctx, tableName)
+			require.NoError(t, err)
+			require.Equal(t, loadTableStat.RowsInserted, int64(14))
+			require.Equal(t, loadTableStat.RowsUpdated, int64(0))
+
+			records := whth.RetrieveRecordsFromWarehouse(
+				t,
+				rs.DB.DB,
+				fmt.Sprintf(`
+					SELECT
+					  id,
+					  received_at,
+					  test_bool,
+					  test_datetime,
+					  test_float,
+					  test_int,
+					  test_string
+					FROM
+					  %q.%q
+					ORDER BY
+					  id ASC;
+				`,
+					warehouse.Namespace,
+					tableName,
+				),
+			)
+			require.Equal(t, records, whth.SampleTestRecords())
+		})
+		t.Run("multiple files", func(t *testing.T) {
+			testCases := []struct {
+				name             string
+				loadByFolderPath bool
+			}{
+				{name: "loadByFolderPath = false", loadByFolderPath: false},
+				{name: "loadByFolderPath = true", loadByFolderPath: true},
+			}
+
+			for i, tc := range testCases {
```

Comment:
Lets add this assertion here as well https://github.com/rudderlabs/rudder-server/pull/5445#discussion_r1930290025